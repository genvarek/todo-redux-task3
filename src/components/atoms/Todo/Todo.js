import React from 'react'
import PropTypes from 'prop-types'

const icon = '\u2716'
const Todo = ({ onClick, text }) => (

  <li>
    <p>{text}</p><button onClick={onClick}>{icon}</button>
  </li>
)

Todo.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Todo

