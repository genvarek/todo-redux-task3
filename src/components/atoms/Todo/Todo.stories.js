import React from 'react'
import { storiesOf } from '@storybook/react'
import Todo from './Todo'

storiesOf('Todo', module)
  .add('default', () => (
    <Todo>Hello</Todo>
  ))
  .add('reverse', () => (
    <Todo reverse>Hello</Todo>
  ))
