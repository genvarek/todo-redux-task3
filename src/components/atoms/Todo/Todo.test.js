import React from 'react'
import { shallow } from 'enzyme'
import Todo from './Todo'

const props = {
  onClick: jest.fn(),
  text: 'kek',
}
const wrap = shallow(<Todo {...props} />)

describe('test', () => {
  it('renders children when passed in', () => {
    wrap.find('button').simulate('click')
    expect(props.onClick).toHaveBeenCalled()
  })

  it('renders props when passed in', () => {
    expect(wrap.contains('kek')).toEqual(true)
  })

  it('to match snapshot', () => {
    expect(wrap).toMatchSnapshot()
  })
})
