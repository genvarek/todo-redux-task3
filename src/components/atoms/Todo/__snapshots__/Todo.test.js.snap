// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    text="kek"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          kek
        </p>,
        <button
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
        >
          ✖
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "kek",
        },
        "ref": null,
        "rendered": "kek",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "✖",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
          },
        },
        "ref": null,
        "rendered": "✖",
        "type": "button",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            kek
          </p>,
          <button
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
          >
            ✖
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "kek",
          },
          "ref": null,
          "rendered": "kek",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "✖",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
            },
          },
          "ref": null,
          "rendered": "✖",
          "type": "button",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
